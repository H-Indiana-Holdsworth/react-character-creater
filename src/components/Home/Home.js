import React, { useState } from 'react';

import Editor from '../Editor/Editor';
import Preview from '../Preview/Preview';

import './Home.css';

export default function Home() {
  const [head, setHead] = useState('head');
  const [torso, setTorso] = useState('torso');
  const [legs, setLegs] = useState('legs');

  return (
    <main>
      <Preview {...{ head, torso, legs }} />
      <Editor {...{ head, setHead, torso, setTorso, legs, setLegs }} />
    </main>
  );
}
