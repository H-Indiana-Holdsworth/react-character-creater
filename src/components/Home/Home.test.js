import { render } from '@testing-library/react';
import Home from './Home';

test('renders Home view', () => {
  const container = render(<Home />);
  expect(container).toMatchSnapshot();
});

test('renders headCount', () => {
  const component = render(<Home headCount torsoCount />);
  expect(component).toMatchSnapshot();
});
