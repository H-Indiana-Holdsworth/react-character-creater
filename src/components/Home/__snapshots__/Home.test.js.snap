// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Home view 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main>
        <div
          class="character"
        >
          <img
            src="/assets/vader-head.jpeg"
          />
          <img
            src="/assets/black-torso.jpeg"
          />
          <img
            src="/assets/black-legs.jpeg"
          />
        </div>
        <div
          class="editor"
        >
          <select>
            <option
              value="vader-head"
            >
              Darth Vader
            </option>
            <option
              value="trooper-head"
            >
              Stormtrooper
            </option>
            <option
              value="r2-head"
            >
              R2-D2
            </option>
          </select>
          <select>
            <option
              value="black-torso"
            >
              Dark torso
            </option>
            <option
              value="red-torso"
            >
              Red torso
            </option>
            <option
              value="woman-torso"
            >
              Woman torso
            </option>
          </select>
          <select>
            <option
              value="black-legs"
            >
              Dark legs
            </option>
            <option
              value="lego-legs"
            >
              Lego legs
            </option>
            <option
              value="white-legs"
            >
              Light legs
            </option>
          </select>
          <input
            type="text"
            value=""
          />
          <button>
            Save
          </button>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main>
      <div
        class="character"
      >
        <img
          src="/assets/vader-head.jpeg"
        />
        <img
          src="/assets/black-torso.jpeg"
        />
        <img
          src="/assets/black-legs.jpeg"
        />
      </div>
      <div
        class="editor"
      >
        <select>
          <option
            value="vader-head"
          >
            Darth Vader
          </option>
          <option
            value="trooper-head"
          >
            Stormtrooper
          </option>
          <option
            value="r2-head"
          >
            R2-D2
          </option>
        </select>
        <select>
          <option
            value="black-torso"
          >
            Dark torso
          </option>
          <option
            value="red-torso"
          >
            Red torso
          </option>
          <option
            value="woman-torso"
          >
            Woman torso
          </option>
        </select>
        <select>
          <option
            value="black-legs"
          >
            Dark legs
          </option>
          <option
            value="lego-legs"
          >
            Lego legs
          </option>
          <option
            value="white-legs"
          >
            Light legs
          </option>
        </select>
        <input
          type="text"
          value=""
        />
        <button>
          Save
        </button>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders headCount 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main>
        <div
          class="character"
        >
          <img
            src="/assets/vader-head.jpeg"
          />
          <img
            src="/assets/black-torso.jpeg"
          />
          <img
            src="/assets/black-legs.jpeg"
          />
        </div>
        <div
          class="editor"
        >
          <select>
            <option
              value="vader-head"
            >
              Darth Vader
            </option>
            <option
              value="trooper-head"
            >
              Stormtrooper
            </option>
            <option
              value="r2-head"
            >
              R2-D2
            </option>
          </select>
          <select>
            <option
              value="black-torso"
            >
              Dark torso
            </option>
            <option
              value="red-torso"
            >
              Red torso
            </option>
            <option
              value="woman-torso"
            >
              Woman torso
            </option>
          </select>
          <select>
            <option
              value="black-legs"
            >
              Dark legs
            </option>
            <option
              value="lego-legs"
            >
              Lego legs
            </option>
            <option
              value="white-legs"
            >
              Light legs
            </option>
          </select>
          <input
            type="text"
            value=""
          />
          <button>
            Save
          </button>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main>
      <div
        class="character"
      >
        <img
          src="/assets/vader-head.jpeg"
        />
        <img
          src="/assets/black-torso.jpeg"
        />
        <img
          src="/assets/black-legs.jpeg"
        />
      </div>
      <div
        class="editor"
      >
        <select>
          <option
            value="vader-head"
          >
            Darth Vader
          </option>
          <option
            value="trooper-head"
          >
            Stormtrooper
          </option>
          <option
            value="r2-head"
          >
            R2-D2
          </option>
        </select>
        <select>
          <option
            value="black-torso"
          >
            Dark torso
          </option>
          <option
            value="red-torso"
          >
            Red torso
          </option>
          <option
            value="woman-torso"
          >
            Woman torso
          </option>
        </select>
        <select>
          <option
            value="black-legs"
          >
            Dark legs
          </option>
          <option
            value="lego-legs"
          >
            Lego legs
          </option>
          <option
            value="white-legs"
          >
            Light legs
          </option>
        </select>
        <input
          type="text"
          value=""
        />
        <button>
          Save
        </button>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
